### ISSUE TITLE
🎅 🎉 Pico Admin Preview Release

### ISSUE THREAD

**Issue by PhrozenByte** [picocms/Pico#378]
It was exactly one year ago that we released Pico 1.0. Unfortunately Pico 2.0 isn't ready yet, but I think that it might get a nice tradition to release at least *something* on Christmas Eve. Thus I'm happy to announce the first, just-a-preview pre-alpha release of Pico's **brand new admin plugin**!

Click here to download the preview: [`pico-admin-preview.tar.gz`](https://github.com/picocms/Pico/files/671863/pico-admin-preview.tar.gz)

This archive contains a full installation of the current development build of Pico 2.0 (see #334) with all necessary dependencies. Navigate to http://example.com/pico/admin and follow the on-screen instructions to use Pico Admin.

You can find `pico-admin`'s source code at https://github.com/PhrozenByte/pico-admin

Merry Christmas and a Happy New Year 2017! 🎅 🎉

@picocms @smcdougall @theshka

**Comment by smcdougall** [comment]
Looks pretty good.  It's got the girlfriend approval factor, lol.  She says she'd probably use that over editing text files directly. (I'd probably stick to text files, but that's just a personal preference :wink:)

I like the password hash generator, it's a nice touch.  Probably a huge usability improvement too (especially over the old readme instructions of the Editor plugins).

From my brief look at it, everything seems really simple, polished, and well thought out.  It seems straightforward enough that anyone could just sit down and use it.  That's a huge plus, as you know how I'm always arguing the case of Pico for the non-techie user.

Also, sorry for the radio-silence lately.  Besides the holidays, I've also been getting ready to move at the end of the month (so... in just a few short days).  It's been really hectic, but everything should be back to business-as-usual in a few weeks.

---

**Merry Christmas, Happy Holidays, and Happy New Year** to you and everyone in the Pico community!

@PhrozenByte Thank you for all the hard work you've put into Pico this year.  Looking forward to the 2.0 release (whenever it's ready).

Let's plan for a big push toward growing Pico and its community throughout 2017!

**Comment by dkyme** [comment]
Happy New Year to you all!

@PhrozenByte thank you for the preview, I'm looking forward to the new Version of Pico. Specially the possibility of data files let me try the preview, buuuut.... I couldn't make it work.

`Moved Temporarily
The server encountered an internal error or misconfiguration and was unable to complete your request.
...
Additionally, a 302 Found error was encountered while trying to use an ErrorDocument to handle the request.
`
showed up after I generated the password (in my case 1234) and posted 
`$config['PicoAdmin']['auth_token'] = '$2y$12$YVdIOfOsXYCdm/MWPtHqC.rKuukJLOp6Ljqp.MudY7seH5mPbx/Hy';
`
into confi.php at the plugin section and clicked on the link.

Manually i can go to admin/content/edit/index and everything is working nicely - Thanks again!!
Also, directly going to admin/login filling in the password, nothing happens, no redirect which I assumed.

But all in all, thank you for the new PlugIn, letting me learn a lot on how to do things in Pico and Twig ... and of course thank you, @smcdougall and @theshka for the effort you put in Pico.

Yours,
Cornelius

**Comment by PhrozenByte** [comment]
@dkyme, sorry for the late response, thanks for testing! :smiley: Very strange issue. What browser and webserver software are you using?

**Comment by dkyme** [comment]
@PhrozenByte, no problem.
Browser is Chrome latest Version 55, and Webserver both Apache, one Synology and a shared Host.
But on shared Host the new .htaccess didn't work at all, I could not get clean Urls at all, only ?sub/page

**Comment by PhrozenByte** [comment]
@dkyme: Have you checked your webserver's `error.log` and a possibly existing separate PHP log?

**Comment by dkyme** [comment]
@PhrozenByte: the error.log of synology complains:

>  comm with server "/php-fpm-handler" aborted: error parsing headers: malformed header '307 Temporary Redirect', referer: http://192.168.0.111/pico2/admin

So far I failed setting up a separate PHP log on my shared host, since I'm more an Copy'n'PHPaster :-(

**Comment by PhrozenByte** [comment]
Sure, that's fine @dkyme, you don't need a separate PHP error log, I'm mentioning this only because people must look at all possible places where PHP errors can show up. In your case the webserver's `error.log` already shows everything we need. This was a bug in Pico Admin, I've fixed it. Thanks for reporting! 👍

**Comment by dkyme** [comment]
Thanks @PhrozenByte ... that works wonderful on my Synology NAS!
On shared Host the URL redirect doesn't works properly.

/?sub works - /sub not
/?sub/page AND /sub/page/ works - /sub/page Not
/?admin redirects to /admin/content/edit/index but shows frontpage

if I use the .htaccess of Version 1.0.4 everything works fine.

**Comment by PhrozenByte** [comment]
@dkyme: Again, very strange issue. Can you please add the following code to line 347 in your `lib/Pico.php` (after the `$this->triggerEvent('onRequestFile', ...` line)?

```php
var_dump($this->requestUrl, $this->requestFile, $_SERVER, $this->config); die();
```

Then navigate to the following pages and give me the output (you might want to use pastebin or something similar). When a error page of your webserver shows up, simply tell me what exact error is shown and check your webserver's `error.log`.

* /
* /?sub/page
* /sub/page/
* /?admin
* /admin
* /?admin/content/edit/index
* /admin/content/edit/index

**Comment by PhrozenByte** [comment]
> Is it an expected behaviour that navigating to /admin the redirect immediatly goes to /admin/content/edit/index while showing the login?

Yes, this is expected since there's currently just one admin module installed.

**Comment by rxng** [comment]
this looks really great! is there support for per user permissions?

**Comment by PhrozenByte** [comment]
@rxng: No, there are also no plannings to add such a feature. It definitely is possible (Pico's plugin system is very powerful), however, things like a role system are explicitly out of Pico's scope.

**Comment by Chris2011** [comment]
@PhrozenByte is there a plan, when this will be released? I tried Pico Edit, it is ok but your implementation looks much better.

**Comment by PhrozenByte** [comment]
@Chris2011 Unfortunately not, "it's done when it's done" 😞 Releasing Pico 2.0 is more important right now and my time is very limited.

**Comment by Chris2011** [comment]
ok, thx for the info :)

**Comment by filisko** [comment]
Is this the roadmap for the version 2? https://github.com/picocms/Pico/pull/334

**Comment by PhrozenByte** [comment]
It initially was, but isn't anymore. However, it will be the next project right after releasing Pico 2.0.

**Comment by filisko** [comment]
@PhrozenByte Oh, I see now.

I will have a look at the left tasks and see if I can help at any.

Thank you @PhrozenByte !

**Comment by PhrozenByte** [comment]
One of the more important tasks is updating Pico's changelog (including a "What has changed" page), writing upgrade instructions and updating the docs (including the install instructions, users should now use a `composer`-based installation when possible). Pico "as is" (i.e. the source code) is basically ready for release.

**Comment by filisko** [comment]
That's good to know, I liked the idea of having official plugins with basic functionality, I'm sure it will help to get more people used to it and be able to use it in a bit more user-friendly way. 👍

**Comment by jens1o** [comment]
#334

**Comment by AndiLeni** [comment]
@jens1o 
All right,
according to the checklist, it's not finished. 
I don't see any information about whether the project was moved, outsourced or cancelled.
Correct me if I'm wrong.

**Comment by PhrozenByte** [comment]
Pico Admin was deferred in favour of Pico 2.0. Pico 2.0 is feature-complete, but the website wasn't updated yet due to limited time. Help is appreciated.

**Comment by AndiLeni** [comment]
I don't know how this might help, but Baun CMS, which is based on Pico, has an interesting approach that seems to be successful throughout. 
See: https://github.com/BaunCMS/Baun
Maybe it is suitable to take up a few ideas :)

**Comment by TheTechRobo** [comment]
Is Pico Admin still being developed? Last commit was a while ago.

**Comment by PhrozenByte** [comment]
Pico Admin is on hold and isn't in development at the moment. It isn't cancelled, just deferred; time unfortunately is very limited and Pico Admin isn't the top priority project at this point.

**Comment by ILDaviz** [comment]
@PhrozenByte  It's interesting add meta_ data and robots on head md file ex:

Title:
Description:
Meta_Description:
Meta_Title: 
Robots:

For seo like.

**Comment by ILDaviz** [comment]
@PhrozenByte It's interesting add robots and menu_presence on head md file ex:
Robots: Index,Follow
Menu_presence: true
For seo like.
